### PR TITLE
feat: Add lhapdf6, gzip, and mg5mes support to PYTHIA8 build 

### LINKED ISSUE #37: Possible to provide `pythia8-config`?
While `pythia8` provides the Pythia header and library files, it doesn't provide the `pythia8-config` command that you get when you build Pythia from source.

```console
$ docker run --rm -ti matthewfeickert/pythia-python:pythia8.308 
docker@6634be5ff711:~/work$ pythia8-config 
Usage: ./pythia8-config [OPTIONS]

Configuration tool for the PYTHIA event generator library. The available 
options are defined below. All available options (without arguments) for the 
PYTHIA configuration script are also valid for this script. See 
"./configuration --help" in the top PYTHIA 8 directory for more details. 

Available options.
--help         : Print this help message (also -h, --h, and -help).
--config       : Print the argument(s) used to configure Pythia.
--prefix       : Installation prefix (cf. autoconf). Note that if the 
                 installation is spread over multiple directories, the
                 binary directory with the trailing "bin" removed is
                 then returned.
--bindir       : PYTHIA binary directory (equivalent to --prefix-bin).
--libdir       : PYTHIA library directory (equivalent to --prefix-lib).
--includedir   : PYTHIA header directory (equivalent to --prefix-include).
--datadir      : PYTHIA share directory (equivalent to --prefix-share).
--xmldoc       : PYTHIA xmldoc directory.
--cxxflags     : Returns the PYTHIA -I flag string needed for compilation.
--cflags       : Equivalent to --cxxflags.
--ldflags      : Returns the PYTHIA -L/-l flag string needed for compilation.
--libs         : Equivalent to --ldflags.
--PACKAGE      : Provides the -I/-L/-l flags needed to link with an external
                 PACKAGE from the following list.
--with-PACKAGE : Returns "true" if the package is enabled, otherwise "false".
  evtgen   : Particle decays with the EvtGen decay pacakge, requires HEPMC2.
  fastjet3 : Building of jets using the FastJet package, version 3.
  	     (run fastjet-config --prefix to see which path to use.)
  hepmc2   : Export PYTHIA events to the HEPMC format, version 2.
  hepmc3   : Export PYTHIA events to the HEPMC format, version 3.
  lhapdf5  : Support the use of external PDF sets via LHAPDF, version 5.
  lhapdf6  : Support the use of external PDF sets via LHAPDF, >= version 6.2.
  powheg   : Hard process production with POWHEGBOX matrix element executables.
  rivet    : Support use of RIVET through direct interface.
  root     : Use ROOT trees and histograms with PYTHIA.
             Note: this option automatically invokes DIR/bin/root-config to set 
             the ROOT lib/ and include/ paths. To set your ROOT paths manually
             instead, use --with-root-lib=DIR and --with-root-include=DIR.
  gzip     : Enable reading of GZIPPED files using the libz library.
  python   : Interface to use PYTHIA in Python.
  mg5mes   : MadGraph matrix element plugins for parton showers.
  openmp   : Multi-threading support via OpenMP.
docker@6634be5ff711:~/work$
```

This is useful for telling other tools about `pythia`. Is it possible to provide this with `pythia8`? If not, what should the plan be for telling other tools about a `pythia8` install? Or is this not needed / already taken care of?

cc @chrisburr @henryiii

### PR DESCRIPTION
**Pull request by matthewfeickert**
Resolves #37

Requires PR #38 to go in first and then should be rebased.

* The options `--enable-shared` and `--enable-64bit` are not supported in PYTHIA 8.310, so they are removed. c.f. https://gitlab.com/Pythia8/releases/-/blob/b5a3fb48f79b3b4441cbf22f7ab710be3b1d7a46/configure
* Bump the build number.
* Add `lhapdf` and `zlip` to host requirements.
* Enable `--with-lhadpdf6`, `--with-gzip`, and `--with-mg5mes` configure options for the PYTHIA8 build.
* MNT: Re-rendered with conda-build 24.7.1, conda-smithy 3.38.0, and conda-forge-pinning 2024.08.13.15.51.23

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by matthewfeickert** [comment]
@conda-forge-admin, please rerender

**Comment by matthewfeickert** [comment]
@conda-forge/pythia8 This is ready for review.

File: recipe/build.sh
Comment (matthewfeickert) [inline recipe/build.sh:14]: The options `--enable-shared` and `--enable-64bit` are not supported in PYTHIA 8.310. c.f. https://gitlab.com/Pythia8/releases/-/blob/b5a3fb48f79b3b4441cbf22f7ab710be3b1d7a46/configure

Though from https://github.com/conda-forge/pythia8-feedstock/issues/37#issuecomment-2287155180 the `.so` files are getting built, so this isn't an issue:

```console
$ find $CONDA_PREFIX -type f -iname "*pythia*.so"
/home/feickert/micromamba/envs/local-build-pythia-debug/lib/pythia8.so
/home/feickert/micromamba/envs/local-build-pythia-debug/lib/libpythia8mg5decays_ew_sm-ckm.so
/home/feickert/micromamba/envs/local-build-pythia-debug/lib/libpythia8lhapdf6.so
/home/feickert/micromamba/envs/local-build-pythia-debug/lib/libpythia8mg5procs_higgs_heft-ckm.so
/home/feickert/micromamba/envs/local-build-pythia-debug/lib/libpythia8mg5procs_top_sm.so
/home/feickert/micromamba/envs/local-build-pythia-debug/lib/libpythia8mg5procs_qcd_sm.so
/home/feickert/micromamba/envs/local-build-pythia-debug/lib/libpythia8.so
/home/feickert/micromamba/envs/local-build-pythia-debug/lib/libpythia8mg5decays_higgs_heft-ckm.so
/home/feickert/micromamba/envs/local-build-pythia-debug/lib/libpythia8mg5procs_dis_sm-ckm.so
/home/feickert/micromamba/envs/local-build-pythia-debug/lib/libpythia8mg5decays_top_sm-ckm.so
/home/feickert/micromamba/envs/local-build-pythia-debug/lib/libpythia8mg5procs_ew_sm-ckm.so
```

File: recipe/meta.yaml
Comment (matthewfeickert) [inline recipe/meta.yaml:36]: I added these to `host` but not `run` as they are required to build with the extensions but you could feasibly run without needing LHAPDF and just optionally install it if you wanted it.

**Review by henryiii — APPROVED** [review]

**Comment by matthewfeickert** [comment]
@henryiii after merge the `linux linux_64_python3.12.____cpython` job failed 4 seconds into the build for some reason unrelated to the code (https://github.com/conda-forge/pythia8-feedstock/runs/28780312155). Is it possible to restart individual jobs on the conda-forge CI?

**Comment by henryiii** [comment]
Sure. Would you like me to add you as a maintainer, BTW?

**Comment by matthewfeickert** [comment]
> Sure.

Thanks!

> Would you like me to add you as a maintainer, BTW?

Sure. I'll add myself in PR #41 that I'm about to rebase.

**Comment by henryiii** [comment]
Okay, sounds good.

File: recipe/meta.yaml
Comment (chrisburr) [reply recipe/meta.yaml:36]: You shouldn't put a version constraint here on zlib. It's included in the global pinning: https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/9e690262e1bb221e8fc8b370232a0aaf7ef1eb2b/recipe/conda_build_config.yaml#L906-L907

When you rerender the versions are picked from there and included in the `.ci_support` files. The reason for defining it there is that if there is a zlib 2 with a different ABI, a "migration" can be ran to rebuild all dependent packages in order with the new zlib. You can track current migrations on the status page: https://conda-forge.org/status/#migrations

It could also make sense to add pythia and lhapdf to the global pinning but I'll leave that up to you.

Comment (matthewfeickert) [reply recipe/meta.yaml:36]: Ah thanks @chrisburr. I didn't know that https://github.com/conda-forge/conda-forge-pinning-feedstock even existed. I'm also just now learning about https://conda-forge.org/status/.

I'll fix that up in a PR following PR #41 and I'll get your sign off there before merging.

> It could also make sense to add pythia and lhapdf to the global pinning but I'll leave that up to you.

Ah, so you're saying that if we added https://github.com/conda-forge/lhapdf-feedstock to the global pinning that anytime there would be a new version of `lhapdf` out then when the `lhapdf` feedstock would get bumped and rebuilt that would trigger a migration that would then also rebuild the pythia8-feedstock?

**Comment by matthewfeickert** [comment]
Ah whoops, seems that this didn't properly fix `--with-gzip`:

```console
$ micromamba list | head -n 4 && micromamba list | grep pythia8
List of packages in environment: "/root/micromamba/envs/local-build-pythia-debug"

  Name                                  Version   Build               Channel    
───────────────────────────────────────────────────────────────────────────────────
  pythia8                               8.310     py312hca68cad_2     conda-forge
$ pythia8-config --with-lhapdf6 --with-gzip --with-python --with-mg5mes
true false true true
```

PR #41 is already in so I won't fix this for PYTHIA 8.310 but I'll make sure to fix this for PYTHIA 8.311 and add some tests to catch this. I might do this in a separate PR following the one to remove the `zlib` lower bound that @chrisburr caught in https://github.com/conda-forge/pythia8-feedstock/pull/39#discussion_r1717462110.

File: recipe/meta.yaml
Comment (chrisburr) [reply recipe/meta.yaml:36]: Anytime there is a version that is deemed ABI incompatible (as defined by the run_exports pattern of lhapdf) a PR like https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/6215 will be opened.

If you want to do the migration, you can ask someone from core to merge it (including me) which will prompt the bots to issue PRs to dependent packages in dependency order.

When all of them are done, a PR like https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/6275 is opened to update the version in the global pinning file.

**Pull request merged**
